src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+
+            final int locusStart = evidence.getContigStart();
+            final int locusEnd = evidence.getContigEnd();
+            final int staleEnd = locusStart - staleEventDistance;
+            int evidenceCount = 0;
+            reportableEntries.clear();
+            final Iterator<Map.Entry<BreakpointEvidence, Boolean>> itr = locMap.entrySet().iterator();
+            while ( itr.hasNext() ) {
+                final Map.Entry<BreakpointEvidence, Boolean> entry = itr.next();
+                final BreakpointEvidence evidence2 = entry.getKey();
+                final int contigEnd = evidence2.getContigEnd();
+                if ( contigEnd <= staleEnd ) itr.remove();
+                else if ( evidence2.getContigStart() >= locusEnd ) break;
+                else if ( contigEnd > locusStart ) {
+                    evidenceCount += 1;
+                    if ( entry.getValue() ) reportableEntries.add(entry);
+                }
+            }
+
+            if ( evidenceCount >= MIN_EVIDENCE ) {
+                return reportableEntries.stream()
+                        .map(entry -> { entry.setValue(false); return entry.getKey(); })
+                        .iterator();
+            }
+            return noEvidence;
+        }
+    }
+
+    /**
+     * Class to fully sort a stream of nearly sorted BreakpointEvidences.
+     */
+    private static final class WindowSorter
+            implements Function<BreakpointEvidence, Iterator<BreakpointEvidence>> {
+        private final SortedSet<BreakpointEvidence> recordSet = new TreeSet<>();
+        private final List<BreakpointEvidence> reportableEvidence = new ArrayList<>();
+        private final int windowSize;
+        private int currentContig = -1;
+
+        WindowSorter( final int windowSize ) {
+            this.windowSize = windowSize;
+        }
+
+        public Iterator<BreakpointEvidence> apply( final BreakpointEvidence evidence ) {
+            reportableEvidence.clear();
+            if ( evidence.getContigIndex() != currentContig ) {
+                reportableEvidence.addAll(recordSet);
+                recordSet.clear();
+                currentContig = evidence.getContigIndex();
+            } else {
+                final int reportableEnd = evidence.getContigStart() - windowSize;
+                final Iterator<BreakpointEvidence> itr = recordSet.iterator();
+                while ( itr.hasNext() ) {
+                    final BreakpointEvidence evidence2 = itr.next();
+                    if ( evidence2.getContigStart() >= reportableEnd ) break;
+                    reportableEvidence.add(evidence2);
+                    itr.remove();
+                }
+            }
+            recordSet.add(evidence);
+            return reportableEvidence.iterator();
+        }
+    }
+
+    /**
+     * Minimalistic simple interval.
+     */
+    @DefaultSerializer(Interval.Serializer.class)
+    @VisibleForTesting static final class Interval {
```

Comment:
What's the reason we can't use `SimpleInterval` here? They hold the same data fields.